src/data.rs

```diff
@@ -0,0 +1,1190 @@
+///   FunctionTemplate Child   -> Child()  . prototype -> { }
+/// ```
+///
+/// A FunctionTemplate 'Child' inherits from 'Parent', the prototype
+/// object of the Child() function has __proto__ pointing to the
+/// Parent() function's prototype object. An instance of the Child
+/// function has all properties on Parent's instance templates.
+///
+/// Let Parent be the FunctionTemplate initialized in the previous
+/// section and create a Child FunctionTemplate by:
+///
+/// ```ignore
+///   Local<FunctionTemplate> parent = t;
+///   Local<FunctionTemplate> child = FunctionTemplate::New();
+///   child->Inherit(parent);
+///
+///   Local<Function> child_function = child->GetFunction();
+///   Local<Object> child_instance = child_function->NewInstance();
+/// ```
+///
+/// The Child function and Child instance will have the following
+/// properties:
+///
+/// ```ignore
+///   child_func.prototype.__proto__ == function.prototype;
+///   child_instance.instance_accessor calls 'InstanceAccessorCallback'
+///   child_instance.instance_property == 3;
+/// ```
+#[repr(C)]
+pub struct FunctionTemplate(Opaque);
+impl<'sc> From<Local<'sc, FunctionTemplate>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, FunctionTemplate>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, FunctionTemplate>> for Local<'sc, Template> {
+  fn from(l: Local<'sc, FunctionTemplate>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// An ObjectTemplate is used to create objects at runtime.
+///
+/// Properties added to an ObjectTemplate are added to each object
+/// created from the ObjectTemplate.
+#[repr(C)]
+pub struct ObjectTemplate(Opaque);
+impl<'sc> From<Local<'sc, ObjectTemplate>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, ObjectTemplate>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, ObjectTemplate>> for Local<'sc, Template> {
+  fn from(l: Local<'sc, ObjectTemplate>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// A compiled JavaScript module, not yet tied to a Context.
+#[repr(C)]
+pub struct UnboundModuleScript(Opaque);
+impl<'sc> From<Local<'sc, UnboundModuleScript>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, UnboundModuleScript>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// The superclass of all JavaScript values and objects.
+#[repr(C)]
+pub struct Value(Opaque);
+impl<'sc> From<Local<'sc, Value>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, Value>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+
+/// A JavaScript value that wraps a C++ void*. This type of value is mainly used
+/// to associate C++ data structures with JavaScript objects.
+#[repr(C)]
+pub struct External(Opaque);
+impl<'sc> From<Local<'sc, External>> for Local<'sc, Data> {
+  fn from(l: Local<'sc, External>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
+impl<'sc> From<Local<'sc, External>> for Local<'sc, Value> {
+  fn from(l: Local<'sc, External>) -> Self {
+    unsafe { transmute(l) }
+  }
+}
```

Comment:
lots of repeated code here - it would be useful if there was a macro so we don't inadvertently make a small error when updating